### PR TITLE
Dump dotenv more smarty

### PR DESCRIPTION
**Pull request by dtan4**
## WHY

Sometimes we want to switch a certain variable temporarily. We use dotenv and flip variables by comment-prefix (`#`) so far.

### normal

```
FOO=bar
DATABASE_URL=postgres://dev.example.com/dev

# ===
#DATABASE_URL=postgres://prod.example.com/prod
```

### connect to production database temporarily

```
FOO=bar
DATABASE_URL=postgres://dev.example.com/dev

# ===
DATABASE_URL=postgres://prod.example.com/prod
```

We will use this method from now on.

By the way, we also use Valec, `valec dump` to _update_ newly-added variable. Currently, `valec dump` overwrites existing `.env` entirely. It is troublesome to set production database URL again.

## WHAT

Add `--output` flag to specify file to dump, and introduce new feature to preserve separated section.

If `.env` already exists and has _separator_, `# ===`, `#---` and so on, the below lines are preserved.

for example:

```bash
$ cat .env
FOO=bar
DATABASE_URL=postgres://dev.example.com/dev

# ===
DATABASE_URL=postgres://prod.example.com/prod
```

```basj
# Added AWESOME_API_KEY
$ valec dump dev --output .env
$ cat .env
FOO=bar
DATABASE_URL=postgres://dev.example.com/dev
AWESOME_API_KEY=dev1234abcd

# ===
DATABASE_URL=postgres://prod.example.com/prod
```

Aside from this, `-o` flag, which is the abbreviation of `--override` is deprecated and now `-o` is used for the abbreviation of `--output`.

### CODE REVIEW

File: util/util.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	separatorRegExp = regexp.MustCompile(`^#+ *(?:-+|=+).*$`)
```
Comment (creasty) [inline util/util.go]: It matches with comments like this:

```sh
###
# Header
###
```

So my suggestion is to change the pattern to `^#+\s*[-=]{3,}$`

Comment (dtan4) [reply util/util.go]: Your pattern will not accept like this header:

```
# ===== production =====
```

How about this?

Comment (creasty) [reply util/util.go]: Indeed. So then `^#+\s*[-=]{3,}` will do, huh?

Comment (dtan4) [reply util/util.go]: Looks good 👍

**Comment by creasty** [comment]
Nice work 👍

**Pull request merged**